qemu/qmp.go

```diff
@@ -1158,6 +1158,30 @@ func (q *QMP) ExecutePCIVFIOMediatedDeviceAdd(ctx context.Context, devID, sysfsd
 	return q.executeCommand(ctx, "device_add", args, nil)
 }
 
+// isSocketIDSupported returns if the cpu driver supports the socket id option
+func isSocketIDSupported(driver string) bool {
+	if driver == "host-s390x-cpu" || driver == "host-powerpc64-cpu" {
+		return false
+	}
+	return true
+}
+
+// isThreadIDSupported returns if the cpu driver supports the thread id option
+func isThreadIDSupported(driver string) bool {
+	if driver == "host-s390x-cpu" || driver == "host-powerpc64-cpu" {
+		return false
+	}
+	return true
+}
+
+// isDieIDSupported returns if the cpu driver and the qemu version support the die id option
+func (q *QMP) isDieIDSupported(driver string) bool {
+	if q.version.Major > 4 || (q.version.Major == 4 && q.version.Minor >= 1) && driver == "host-x86_64-cpu" {
```

Comment:
Is this test correct?  Won't it always enable die-id for qemu version 5 and above regardless of the driver?

Reply:
good catch!

Reply:
I added the parenthesis, sorry